### PR TITLE
[vec] add GE2E to support unlabeled data training

### LINKED ISSUE #1730: [vec] support unsupervised learning

### PR DESCRIPTION
**Pull request by qingen**
add GE2E to support unlabeled data training by contrastive learning, test=doc fix #1730

### CODE REVIEW

**Comment by yt605155624** [comment]
之前飞宇复现了 GE2E 这个声纹模型，也有相应的 loss, 供参考 https://github.com/PaddlePaddle/PaddleSpeech/blob/f39de8d754c4bf47d9ee0997915a47e9e7e1b6ec/paddlespeech/vector/models/lstm_speaker_encoder.py#L114

**Comment by qingen** [comment]
> 之前飞宇复现了 GE2E 这个声纹模型，也有相应的 loss, 供参考
> 
> https://github.com/PaddlePaddle/PaddleSpeech/blob/f39de8d754c4bf47d9ee0997915a47e9e7e1b6ec/paddlespeech/vector/models/lstm_speaker_encoder.py#L114

嗯，谢谢，这个Loss需要单独抽取出来，并且方法上需要完善，增加比较学习策略

File: paddlespeech/vector/modules/loss.py [hunk 1]
```diff
@@ -268,22 +269,128 @@ def forward(self, outputs, targets):
...
+        # cal loss
+        if self.loss_method == "softmax":
+            loss = self.cal_softmax_loss(cossims)
+        else:
+            loss = self.cal_contrast_loss(cossims)
+
+        return loss
+
+
 if __name__ == "__main__":
```
Comment (zh794390558) [inline paddlespeech/vector/modules/loss.py:371]: 测试统一到tests/unit/vector中

**Review by LeoMax-Xiong — APPROVED** [review]

File: paddlespeech/vector/modules/loss.py (hunk 1 above)
Comment (qingen) [reply paddlespeech/vector/modules/loss.py:371]: get

**Comment by starmoon-1134** [comment]
请教一下，基于对比学习的GE2E效果怎么样呢，看代码好像只在paddlespeech/vector/modules/loss.py测试用了一下，没在GE2E训练流程里使用
